django/db/models/query.py

```diff
@@ -1707,9 +1722,11 @@ def populate(self, row, from_obj):
         if obj and self.related_populators:
             for rel_iter in self.related_populators:
                 rel_iter.populate(row, obj)
-        setattr(from_obj, self.cache_name, obj)
-        if obj and self.reverse_cache_name:
-            setattr(obj, self.reverse_cache_name, from_obj)
+
+        self.field.set_cached_value(instance=from_obj, value=obj)
```

Comment:
Are you passing args as kwargs like this and throughout the patch because of readability? I'm not sure it helps -- it seems natural that a `set()` method would take `(key, value)`.